ui-tests/src/test/java/org/openmrs/reference/LoginTest.java

```diff
@@ -12,19 +12,20 @@
 import org.junit.Ignore;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
+import org.openmrs.reference.groups.BuildTests;
 import org.openmrs.reference.page.HomePage;
 
 import static org.junit.Assert.assertTrue;
 
-@Ignore
+
 public class LoginTest extends ReferenceApplicationTestBase {
     private HomePage homePage;
-
+    
     @Test
-    @Category(org.openmrs.reference.groups.BuildTests.class)
+    @Category(BuildTests.class)
     public void verifyModulesAvailableOnHomePage() throws Exception {
+    	goToLoginPage().loginAsAdmin();
         homePage = new HomePage(page);
-        assertPage(homePage);
```

Comment:
any reasons for removing these _asserPage()_

Reply:
The function is very vital  as it checks the current page see https://github.com/openmrs/openmrs-contrib-uitestframework/blob/master/src/main/java/org/openmrs/uitestframework/test/TestBase.java#L340

Reply:
Sure it checks the current Page using that method but the current page here already by default has been reached , initialising homePage with in the class creates a new instance which is by default homePage, its like a repetition of asserting page(homePage)

Reply:
> the current page here already by default has been reached

What would happen if the page does not exist and the function assertPage() is no longer present

Reply:
Didnot get your point well but ideally  homePage is inherited from loginPage, the login workflow is basing on homePage, whatever we are testing falls within homepage, we cant say after logging in the automation is redirected to back unless its a different test

Reply:
> Didnot get your point well

Take a `use case` when the page does not exist and the function `assertPage()` is no longer present as suggested in the your PR. How will this be handled ?

Reply:
The page will by default exist ofcourse like i said above these pages are paginating to each other, when you say page doesnot exist , it means loginfunction wil not be working

Reply:
As removing `assertPage()` may seem to be the solution, But this will cause a lot of issues in the long run. So I recommend to retain this method within the class.

Reply:
Before you recommend please could you throw tell us which issues do you think it might cause in the long run thanks

Reply:
> please could you tell us which issues do you think it might cause

Refer https://github.com/openmrs/openmrs-contrib-uitestframework/blob/master/src/main/java/org/openmrs/uitestframework/test/TestBase.java#L340

Reply:
ok thank you let me look through that , but thats not the problem i guess , suppose you are asserting the page when you are already on the expected page

Reply:
The `function` is very vital as it checks the current page and it should not be omitted.

Reply:
ok thanks for your review

Reply:
work around ensuring the test passes with the presence of this function rather than getting rid of it.

Reply:
> but thats not the problem i gues
Add `waitForPage()`  function on `homePage` object to ensure the page is loaded for assertion to be effective. Something like `assertPage(homePage..waitForPage());`

Reply:
You have still not getting my point, even though we add a wait element tag, the assertion expection  wont change

Reply:
> even though we add a wait element tag, the assertion expection wont change

Have you taken a use case on this

Reply:
Havent tried waiting for homePage still wondering why waiting for homePage when its throwing assertion error migh change its assertion to pass

Reply:
> Havent tried waiting for homePage

Does it still throw the exception?

Reply:
its working thanks though practically it does not add up

Reply:
> though practically it does not add up

Refer to my recent response [here](https://issues.openmrs.org/browse/RATEST-93)